packages/block-editor/src/components/url-popover/image-url-input-ui.js

```diff
@@ -142,7 +142,15 @@ const ImageURLInputUI = ( {
 	const onSubmitLinkChange = useCallback( () => {
 		return ( event ) => {
 			if ( urlInput ) {
-				onChangeUrl( { href: urlInput } );
+				const selectedDestination =
+					getLinkDestinations().find(
+						( destination ) => destination.url === urlInput
+					)?.linkDestination || LINK_DESTINATION_CUSTOM;
```

Comment:
This check is needed to account for instances where the url entered matches the media or attachment urls

Reply:
That's correct 👍